app/models/user.rb

```diff
@@ -12,17 +12,17 @@ class User < ApplicationRecord
 
   belongs_to :casa_org
 
-  has_many :case_assignments, foreign_key: "volunteer_id"
+  has_many :case_assignments, foreign_key: "volunteer_id", dependent: :destroy
   has_many :casa_cases, through: :case_assignments
-  has_many :case_contacts, foreign_key: "creator_id"
+  has_many :case_contacts, foreign_key: "creator_id", dependent: :destroy
```

Comment:
I don't think we want `dependent: :destroy` on these because we don't want to lose these `case_assignments` and `case_contacts`. So let's do `dependent: :nullify` which will just clear the `volunteer_id` and `creator_id` fields.

Reply:
But actually, nullify will make the corresponding association invalid. Let's just remove the `dependent: ` from here completely. If we ever want to delete a user for reals and we hit a foreign key error, then we will need a more sophisticated approach to reassign this. Does it affect the seeds file if you just remove this change?

Reply:
Assuming a user can be _deactivated_ rather than _deleted_, what would be the reason to delete a user if that user is part of the history of a case that is _not_ deleted? If you want to save the case contact history, don't you also want to save the part of that history that identifies the party who made the contact? Wouldn't it be important for practical and maybe even legal reasons?  I noticed that there is an active flag on the User instance and was assuming it would be used instead of deletion for users with case history.

If we remove the dependent:destroy from here, then as you said, we would be leaving the DB in an invalid state. A workaround might be to have a pseudo-user object to represent deleted users, but I think we're voluntarily forfeiting informational value if we erase that history.

Any class whose objects are not deleted implicitly can easily be deleted explicitly in the seeds file, so we have complete flexibility with the domain models.

I'm fine either way but wanted to mention this first before making the change.

Reply:
@littleforest I've removed the dependent/destroy as you suggested and destroy the CaseContact instances explicitly in the seeder instead. As an aside, I tried dependent/nullify first, and the user deletion failed because there is a not null constraint on the creator field in case contact.

Reply:
Yes, probably the user would be deactivated rather than deleted, but it's not impossible for someone who has access to the production database to delete a user outside of the UI. In that case we want an error thrown, rather than the code deleting these records. So we should remove the `dependent: :destroy` here, and just make sure the seeds file works fine in terms of the order it deletes records, etc.

Reply:
I believe this PR is ready to merge. @compwron @littleforest